std/experimental/allocator/package.d

```diff
@@ -0,0 +1,1790 @@
+    array[oldLength .. $].uninitializedFill(init);
+    return true;
+}
+
+/// Ditto
+bool expandArray(T, Allocator, R)(auto ref Allocator alloc, ref T[] array,
+        R range)
+if (isInputRange!R)
+{
+    static if (isForwardRange!R)
+    {
+        immutable delta = walkLength(range.save);
+        if (!delta) return true;
+        immutable oldLength = array.length;
+
+        // Reallocate support memory
+        void[] buf = array;
+        if (!alloc.reallocate(buf, buf.length + T.sizeof * delta))
+        {
+            return false;
+        }
+        array = cast(T[]) buf;
+        // At this point we're committed to the new length.
+
+        auto toFill = array[oldLength .. $];
+        scope (failure)
+        {
+            // Fill the remainder with default-constructed data
+            toFill.uninitializedFillDefault;
+        }
+
+        for (; !range.empty; range.popFront, toFill.popFront)
+        {
+            assert(!toFill.empty);
+            import std.conv : emplace;
+            emplace!T(&toFill.front, range.front);
+        }
+        assert(toFill.empty);
+    }
+    else
+    {
+        scope(failure)
+        {
+            // The last element didn't make it, fill with default
+            array[$ - 1 .. $].uninitializedFillDefault;
+        }
+        void[] buf = array;
+        for (; !range.empty; range.popFront)
+        {
+            if (!alloc.reallocate(buf, buf.length + T.sizeof))
+            {
+                array = cast(T[]) buf;
+                return false;
+            }
+            import std.conv : emplace;
+            emplace!T(buf[$ - T.sizeof .. $], range.front);
+        }
+
+        array = cast(T[]) buf;
+    }
+    return true;
+}
+
+///
+unittest
+{
+    auto arr = theAllocator.makeArray!int([1, 2, 3]);
+    assert(theAllocator.expandArray(arr, 2));
+    assert(arr == [1, 2, 3, 0, 0]);
+    import std.range : only;
+    assert(theAllocator.expandArray(arr, only(4, 5)));
+    assert(arr == [1, 2, 3, 0, 0, 4, 5]);
+
+    ForcedInputRange r;
+    int[] b = [ 1, 2, 3, 4 ];
+    auto temp = b;
+    r.array = &temp;
+    assert(theAllocator.expandArray(arr, r));
+    assert(arr == [1, 2, 3, 0, 0, 4, 5, 1, 2, 3, 4]);
+}
+
+/**
+Shrinks an array by $(D delta) elements.
+
+If $(D array.length < delta), does nothing and returns `false`. Otherwise,
+destroys the last $(D array.length - delta) elements in the array and then
+reallocates the array's buffer. If reallocation fails, fills the array with
+default-initialized data.
+
+Params:
+T = element type of the array being created
+alloc = the allocator used for getting memory
+array = a reference to the array being shrunk
+delta = number of elements to remove (upon success the new length of $(D array) is $(D array.length - delta))
+
+Returns:
+`true` upon success, `false` if memory could not be reallocated. In the latter
+case, the slice $(D array[$ - delta .. $]) is left with default-initialized
+elements.
+
+Throws:
```

Comment:
There is a copy-paste error. There are no three overloads for shrinkArray.

P. S. Is there a way on github to search a proper place for comment other than scrolling down the whole pull request?